### DISCUSSION TITLE
Automatic "y" for installation via curl?

### DISCUSSION THREAD

**Discussion by jasontheiler** [starship/starship#1574] [Q&A] [answered]
Is there any way to answer the following installation question automatically with "y"?

```bash
  Configuration
> Bin directory: /usr/local/bin
> Platform:      unknown-linux-gnu
> Arch:          x86_64

> Tarball URL: https://github.com/starship/starship/releases/latest/download/starship-x86_64-unknown-linux-gnu.tar.gz
? Install Starship latest to /usr/local/bin? [y/N]
```

I'd like to automatically install Starship with a shellscript so I can get a new WSL2 instance up and running quickly, while keeping the amount of manual interaction to a minimum. I've tried to use `yes | curl -fsSL https://starship.rs/install.sh | bash`, but with no success. I also haven't found any other documentation, issues or discussions related to this.

I absolutely love Starship so far, though! Thanks a ton for this beautiful prompt!

**Comment by matchai** [comment, marked as answer]
The accepted flags are commented at the top of the install script:
https://github.com/starship/starship/blob/master/install/install.sh#L3-L21

You can use the following to pass the flag straight into the `curl | bash`:
```
curl -fsSL https://starship.rs/install.sh | bash -s -- --yes
```

**Reply by jasontheiler** [reply]
Ah, thanks! I didn't think of looking into the install script itself...

It would be nice to have those in the official documentation, though. Maybe as a sub-category in [Installation](https://starship.rs/guide/#%F0%9F%9A%80-installation) or as a question in [Frequently Asked Questions](https://starship.rs/faq).

Anyways, thanks again! 😄

**Comment by goetzc** [comment]
When not having Bash installed, this works to automatically determine the current shell (avoiding a possible parent process) on GNU/Linux and BSD/macOS systems:

```sh
STARSHIP_SHELL="$(ps -hp $$ | awk '{print $5}' | tr -d '-')"
```

Then the usual install step with the initialization for the same shell:

```sh
curl -fsSL https://starship.rs/install.sh | $STARSHIP_SHELL -s -- --yes
eval "$(starship init $STARSHIP_SHELL)"
```

**Reply by davidkna** [reply]
The install script is written in bash so it might not work on other shells (especially non-posixlike shells like powershell and ion).
